### PR TITLE
Remove not_expired scope from for_namespace_and_ancestors scope

### PR DESCRIPTION
**Pull request by deepsidhu85**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR removes `not_expired scope` from `for_namespace_and_ancestors` scope and instead applies it to the collection of for_namespace_and_ancestors when getting the user's access level through group links. The reasoning for this is we will need the scope to be able to display the expired and non expired group links on the Members -> Groups tab

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as a `user1@email.com`
2. Create 2 groups `Group A` and `Group B`
3. Create a namespace group link between 2 groups `Group A` and `Group B` where `Group B` is invited to `Group A`
4. Update the group link to a date in the past
5. Add `user2@email.com` to `Group B`
6. Login as `user2@email.com`
7. Visit the URL for `Group A` and you should get an Access Denied page as the group link has expired

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/models/namespace_group_link.rb [hunk 1]
```diff
@@ -23,7 +23,6 @@ class NamespaceGroupLink < ApplicationRecord
   scope :not_expired, -> { where('expires_at IS NULL OR expires_at > ?', Time.zone.now) }
   scope :for_namespace_and_ancestors, lambda { |namespace = nil|
```
Comment (ericenns) [inline app/models/namespace_group_link.rb:24]: **nitpick (blocking):** Could you please add a test for this scope.

For testing, I recommend that you use a Group with ancestors and use the `shared_with_group_links` association to get all of them and then ensure that they are all present in `for_namespace_and_ancestors`

**Review by ericenns — CHANGES_REQUESTED** [review]
This is looking good, but I would like tests added for the `for_namespace_and_ancestors` scope which was not previously present.

File: app/models/namespace_group_link.rb (hunk 1 above)
Comment (deepsidhu85) [reply app/models/namespace_group_link.rb:24]: Added scope testing for both namespace_group_link and member models in 609ccdf0cad480f8926e7af4e2159dffe1f67a26

File: test/models/namespace_group_link_test.rb [hunk 2]
```diff
@@ -65,4 +65,17 @@ def setup
...
+
+  test '#scope for_namespace_and_ancestors returns the correct collection' do
+    group_group_link = namespace_group_links(:namespace_group_link2)
+
+    group = group_group_link.namespace
+
+    shared_with_group_links = group.shared_with_group_links
+    namespace_group_links = NamespaceGroupLink.for_namespace_and_ancestors(group)
+
+    shared_with_group_links.each do |shared_with_group_link|
```
Comment (ericenns) [inline test/models/namespace_group_link_test.rb:77]: **nitpick (blocking):** Can you add a test to ensure that the counts are the same. That way we know that both collections are the same.

File: test/models/members/project_member_test.rb [hunk 3]
```diff
@@ -126,4 +126,11 @@ def setup
       Member.restore(@project_member.id, recursive: true)
     end
   end
+
+  test '#scope for_namespace_and_ancestors returns the correct collection' do
```
Comment (ericenns) [inline test/models/members/project_member_test.rb:130]: **nitpick (blocking):** Can you update this test to include testing for members from the Group and Ancestors that the project belongs to. So similar to NamespaceGroupLink and also make sure that the total is the same.

**Review by ericenns — CHANGES_REQUESTED** [review]
Just a couple minor testing improvements.

File: test/models/namespace_group_link_test.rb (hunk 2 above)
Comment (deepsidhu85) [reply test/models/namespace_group_link_test.rb]: Updated test to check that the collections and counts are the same in 07d46704b0aab5511c116e8688c9ae977526cd96

File: test/models/members/project_member_test.rb (hunk 3 above)
Comment (deepsidhu85) [reply test/models/members/project_member_test.rb:130]: Updated in 07d46704b0aab5511c116e8688c9ae977526cd96

**Review by ericenns — APPROVED** [review]
Looks good to me!

**Pull request merged**